src/scss/init/unity-layout/_unity-layout.scss

```diff
@@ -0,0 +1,199 @@
+// is-number($value)
+//
+// Returns true if $value variable is of type number
+//
+// Weight: -7
+//
+// Style guide: init.libraries.unity-layout.is-number
+@function is-number($value) {
+  @return type_of($value) == 'number';
+}
+
+// is-integer($value)
+//
+// Returns true if $value variable is an integer
+//
+// Weight: -6
+//
+// Style guide: init.libraries.unity-layout.is-integer
+@function is-integer($value) {
+  @return is-number($value) and round($value) == $value;
+}
+
+// is-percentage($value)
+//
+// Return true if $value variable is a percentage with no decimals
+//
+// Weight: -5
+//
+// Style guide: init.libraries.unity-layout.is-percentage
+@function is-percentage($value) {
+  @return is-integer($value) and unit($value) == '%';
+}
+
+// convert-percentage($value)
+//
+// Check if $value variable is a valid integer or percentage value. If $value is an integer, but not a percentage
+// then return a percentage value. If $value is neither an integer, nor a percentage, then return false.
+//
+// Weight: -4
+//
+// Style guide: init.libraries.unity-layout.convert-percentage
+@function convert-percentage($value) {
+  @if is-integer($value) and not(is-percentage($value)) {
+    $percent-value: $value * 1%;
+    @return $percent-value;
+  } @else if is-percentage($value) {
+    @return $value;
+  } @else {
+    @return false;
+  }
+}
+
+// is-valid($values)
+//
+// Pass each $value in $values list through convert-percentage() function. Return the new list of $values if all values
+// are valid, if not, return false.
+//
+// Weight: -3
+//
+// Style guide: init.libraries.unity-layout.is-valid
+@function is-valid($values) {
```

Comment:
make this a private function for this scss script, also maybe give it a more semantic name to make it clear that its scope is limited to the the unity layout mixin